enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,533 @@
+`/templates/arbiter/**/kubelet.service.yaml`
+
+```
+        --node-labels=node-role.kubernetes.io/arbiter,node.openshift.io/os_id=${ID} \
+...
+        --register-with-taints=node-role.kubernetes.io/arbiter=:NoSchedule \
+```
+
+### Installer Changes
+
+We will need to update the installer to have awareness of the explicit intent to
+setup the cluster with an arbiter node. Adding a new machine type similar to the
+flow for `master` and `workers` machines. The intent will be to generate 1
+`arbiter` machine and 2 `master` machines when the user specifies a
+`MachinePool` `install-config.arbiterNode` field.
+
+The `arbiterNode` will be a `MachinePool` object that will enforce 1 replica,
+and users will supply 2 replicas for `controlPlane`, when no `arbiterNode` field
+is supplied regular flow will validate replica count. When `arbiterNode` is
+supplied 2 replicas will be valid for for `controlPlane` as long as an
+`arbiterNode` is specified.
+
+`installConfig.yaml`
+
+```yaml
+apiVersion: v1
+baseDomain: devcluster.openshift.com
+compute:
+  - architecture: amd64
+    hyperthreading: Enabled
+    name: worker
+    platform: {}
+    replicas: 0
+arbiterNode:
+  platform:
+    aws:
+      type: m6a.xlarge
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  platform:
+    aws:
+      type: m6a.4xlarge
+  replicas: 2
+```
+
+Ex. `/machines/arbiter.go`
+
+```go
+type Arbiter struct {
+	UserDataFile           *asset.File
+	MachineConfigFiles     []*asset.File
+	MachineFiles           []*asset.File
+	ControlPlaneMachineSet *asset.File
+	IPClaimFiles           []*asset.File
+	IPAddrFiles            []*asset.File
+
+	// SecretFiles is used by the baremetal platform to register the
+	// credential information for communicating with management
+	// controllers on hosts.
+	SecretFiles []*asset.File
+
+	// NetworkConfigSecretFiles is used by the baremetal platform to
+	// store the networking configuration per host
+	NetworkConfigSecretFiles []*asset.File
+
+	// HostFiles is the list of baremetal hosts provided in the
+	// installer configuration.
+	HostFiles []*asset.File
+}
+```
+
+### ETCD Operator Change
+
+CEO(cluster-etcd-operator) will need to be made aware of the new arbiter role.
+There are some challenges posed by this since CEO uses the node informer for
+`master` in a lot of their controllers to manage the etcd cluster. We also have
+a challenge in that the `node-role` labels are just empty keys, meaning that a
+label selector like `node-role in (master,arbiter)` will not work for
+`node-role` since we always just match on the key's existence. Furthermore,
+labels selectors by design do not support an `or` operation on keys
+([see](https://github.com/kubernetes/kubernetes/issues/90549#issuecomment-620625847)).
+We will need to wrap the informer and lister interfaces to allow much of the
+logic not to change for CEO, this does also offer a benefit since it will be
+easier to FeatureGate during our TechPreview phase.
```

Comment:
I don't think we're going to feature gate a "joint informer" :) that doesn't seem worth the hassle

Reply:
Lol that's fair, I'll clean up that line.